File: src/main/java/com/javatechie/webflux/handler/CustomerStreamHandler.java
package com.javatechie.webflux.handler;

import com.javatechie.webflux.dao.CustomerDao;
import com.javatechie.webflux.dto.Customer;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.stereotype.Service;
import org.springframework.web.reactive.function.server.ServerRequest;
import org.springframework.web.reactive.function.server.ServerResponse;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Service
public class CustomerStreamHandler {

    @Autowired
    private CustomerDao dao;


    public Mono<ServerResponse> getCustomers(ServerRequest request) {
        Flux<Customer> customersStream = dao.getCustomersStream();
        return ServerResponse.ok().
                contentType(MediaType.TEXT_EVENT_STREAM)
                .body(customersStream, Customer.class);
    }
}

File: springboot-webflux-demo-Kotlin/src/main/kotlin/com/myproject/webflux/handler/CustomerStreamHandler.kt
package com.myproject.webflux.handler

import com.myproject.webflux.dto.Customer

import org.springframework.web.reactive.function.server.ServerResponse

import reactor.core.publisher.Flux

import org.springframework.web.reactive.function.server.ServerRequest

import reactor.core.publisher.Mono

import com.myproject.webflux.dao.CustomerDao

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.MediaType
import org.springframework.stereotype.Service


@Service
class CustomerStreamHandler {
    @Autowired
    private val dao: CustomerDao? = null
    fun getCustomers(request: ServerRequest?): Mono<ServerResponse> {
        val customersStream = dao!!.getCustomersStream()
        return ServerResponse.ok().contentType(MediaType.TEXT_EVENT_STREAM)
            .body(customersStream, Customer::class.java)
    }
}